pkg/config/marshal.go

```diff
@@ -137,3 +119,48 @@ func UnmarshalTLS(val interface{}) TLS {
 		RootCAs:     certPool,
 	}
 }
+
+// Marshal updates the Store with Config dirty fields.
+func Marshal(conf *Config, store *Store) {
```

Comment:
In https://golang.org/pkg/encoding/json/#Marshal, `Marshal` returns a new Store instead of taking it as a parameter and customizes it. Should we do the same?

Reply:
Interesting, it's possible there because they can generate the JSON bytes from scratch. In our case we can't always do so because we might have an already existing store with a path and potential config data that we only need to update.